eth2/operation_pool/src/attestation.rs

```diff
@@ -1,35 +1,53 @@
 use crate::max_cover::MaxCover;
-use types::{Attestation, BeaconState, BitList, EthSpec};
+use state_processing::common::{get_attesting_indices, get_base_reward};
+use std::collections::HashMap;
+use std::convert::TryInto;
+use types::{Attestation, BeaconState, BitList, ChainSpec, EthSpec};
 
 pub struct AttMaxCover<'a, T: EthSpec> {
     /// Underlying attestation.
     att: &'a Attestation<T>,
-    /// Bitfield of validators that are covered by this attestation.
-    fresh_validators: BitList<T::MaxValidatorsPerCommittee>,
+    /// Mapping of validator indices and their rewards.
+    fresh_validators_rewards: HashMap<u64, u64>,
 }
 
 impl<'a, T: EthSpec> AttMaxCover<'a, T> {
     pub fn new(
         att: &'a Attestation<T>,
-        fresh_validators: BitList<T::MaxValidatorsPerCommittee>,
+        state: &BeaconState<T>,
+        spec: &ChainSpec,
+        total_active_balance: u64,
     ) -> Self {
+        let fresh_validators = earliest_attestation_validators(att, state);
+        let indices = get_attesting_indices(state, &att.data, &fresh_validators)
+            .expect("should have returned valid indices");
```

Comment:
It would be good to avoid the `expect` here (_just in case_). Given that we do expect most of these to succeed we could make this function return an `Option<Self>`, and change the `map` call in `lib.rs` to a `flat_map`. That way any attestations with errors (of which there shouldn't be any) will just be excluded from max cover.

Reply:
This expect still needs removing